lldb/test/API/tools/lldb-dap/attach/TestDAP_attachByPortNum.py

```diff
@@ -0,0 +1,146 @@
+"""
+Test lldb-dap "port" configuration to "attach" request
+"""
+
+
+import dap_server
+from lldbsuite.test.decorators import *
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test import lldbutil
+from lldbsuite.test import lldbplatformutil
+import lldbgdbserverutils
+import lldbdap_testcase
+import os
+import shutil
+import subprocess
+import tempfile
+import threading
+import time
+import sys
+
+
+class TestDAP_attachByPortNum(lldbdap_testcase.DAPTestCaseBase):
+    def runTargetProgramOnPort(self, port=None, program=None):
+        server_tool = None
+        if lldbplatformutil.getPlatform() == "linux":
+            server_tool = lldbgdbserverutils.get_lldb_server_exe()
+            if server_tool is None:
+                self.dap_server.request_disconnect(terminateDebuggee=True)
+                self.assertIsNotNone(server_tool, "lldb-server not found.")
+            server_tool += " g localhost:" + port + " "
+        elif lldbplatformutil.getPlatform() == "macosx":
+            server_tool = lldbgdbserverutils.get_debugserver_exe()
+            if server_tool is None:
+                self.dap_server.request_disconnect(terminateDebuggee=True)
+                self.assertIsNotNone(server_tool, "debugserver not found.")
+            server_tool += " --listen localhost:" + port + " "
```

Comment:
could you move the logic that finds automatically the path to lldb-server or debugserver to lldbdap_testcase? This logic seems to be useful to reuse

Reply:
Agreed with your comment.
I think it would be better to move this into lldbtest.py instead of lldb_testcase,py as it has all utility funtions.

Reply:
1. Moved this piece of code into lldbdap_testcase.py: getBuiltinServerToolWithPortArg(self, port):